src/components/ui/CopyLink/CopyLink.tsx

```diff
@@ -0,0 +1,75 @@
+import { AnimatePresence, motion } from 'framer-motion';
+import { useEffect, useState } from 'react';
+
+export const CopyLink = ({ link }: { link: string }) => {
+  const [copied, setCopied] = useState(false);
+
+  // Reset copied state after 1.5 seconds to allow user to copy again
+  useEffect(() => {
+    if (copied) {
+      const timer = setTimeout(() => {
+        setCopied(false);
+      }, 1500);
+      return () => clearTimeout(timer);
+    }
+  }, [copied]);
+
+  const copyLink = () => {
+    navigator.clipboard.writeText(link);
+    setCopied(true);
+  };
+
+  return (
+    <div className="flex items-center gap-[10px] h-10 rounded-full justify-between px-2 pr-4 shadow-[0_0_8px_rgba(4,41,49,0.15)] text-[13px] relative">
+      <div className="whitespace-nowrap overflow-x-clip">{link}</div>
+      <div className="w-14 h-6 bg-gradient-to-l from-white right-[88px] absolute"></div>
```

Comment:
Love this for the gradient! Good job here!

Just a very minimal comment here, which is just a slight improvement, as this element is actually only visible and not intractable for the user, I'd recommend adding the `pointer-events-none` class (docs [here](https://tailwindcss.com/docs/pointer-events)).

Reply:
@Publicker I tried this and it changed the pointer to cursor: text. Without this, it was just a cursor: default. It's not part of the copy link button so it will not inherit the cursor.